termwiz/src/surface/line.rs

```diff
@@ -730,7 +730,7 @@ mod test {
             "❤ 😍🤢 http://example.com \u{1f468}\u{1f3fe}\u{200d}\u{1f9b0} http://example.com";
```

Comment:
it would be neat to put in a URL like `http://localhost:3000` here and adjust the expectations below to see that it matched and turned it into a link!

Reply:
Hmm, just 
```rs
        let text =
            "❤ 😍🤢 http://example.com \u{1f468}\u{1f3fe}\u{200d}\u{1f9b0} http://example.com http://localhost:3000";
```
?

Reply:
Yeah, and then update the test assertions to match.  you can `cargo test` to run the tests

Reply:
I am really not sure how to update the tests tbh, since i don't know much about Rust. My only guess was appending this after L799 (`Cell::new(' ', CellAttributes::default()),`)
```rs
                Cell::new('h', hyperlink_attr.clone()),
                Cell::new('t', hyperlink_attr.clone()),
                Cell::new('t', hyperlink_attr.clone()),
                Cell::new('p', hyperlink_attr.clone()),
                Cell::new(':', hyperlink_attr.clone()),
                Cell::new('/', hyperlink_attr.clone()),
                Cell::new('/', hyperlink_attr.clone()),
                Cell::new('l', hyperlink_attr.clone()),
                Cell::new('o', hyperlink_attr.clone()),
                Cell::new('c', hyperlink_attr.clone()),
                Cell::new('a', hyperlink_attr.clone()),
                Cell::new('l', hyperlink_attr.clone()),
                Cell::new('h', hyperlink_attr.clone()),
                Cell::new('o', hyperlink_attr.clone()),
                Cell::new('s', hyperlink_attr.clone()),
                Cell::new('t', hyperlink_attr.clone()),
                Cell::new(':', hyperlink_attr.clone()),
                Cell::new('3', hyperlink_attr.clone()),
                Cell::new('0', hyperlink_attr.clone()),
                Cell::new('0', hyperlink_attr.clone()),
                Cell::new('0', hyperlink_attr.clone()),
```

Reply:
Update: I made the Tests pass, by adding
```rs
        let hyperlink_port = Arc::new(Hyperlink::new_implicit("http://localhost:3000"));
        let hyperlink_port_attr = CellAttributes::default()
            .set_hyperlink(Some(hyperlink_port.clone()))
            .clone();
```
and
```rs
                Cell::new('h', hyperlink_port_attr.clone()),
                Cell::new('t', hyperlink_port_attr.clone()),
                Cell::new('t', hyperlink_port_attr.clone()),
                Cell::new('p', hyperlink_port_attr.clone()),
                Cell::new(':', hyperlink_port_attr.clone()),
                Cell::new('/', hyperlink_port_attr.clone()),
                Cell::new('/', hyperlink_port_attr.clone()),
                Cell::new('l', hyperlink_port_attr.clone()),
                Cell::new('o', hyperlink_port_attr.clone()),
                Cell::new('c', hyperlink_port_attr.clone()),
                Cell::new('a', hyperlink_port_attr.clone()),
                Cell::new('l', hyperlink_port_attr.clone()),
                Cell::new('h', hyperlink_port_attr.clone()),
                Cell::new('o', hyperlink_port_attr.clone()),
                Cell::new('s', hyperlink_port_attr.clone()),
                Cell::new('t', hyperlink_port_attr.clone()),
                Cell::new(':', hyperlink_port_attr.clone()),
                Cell::new('3', hyperlink_port_attr.clone()),
                Cell::new('0', hyperlink_port_attr.clone()),
                Cell::new('0', hyperlink_port_attr.clone()),
                Cell::new('0', hyperlink_port_attr.clone()),
```
but i am not really sure if that is correct.

Reply:
yeah, that looks about right: please push it and we can let the tests run on it here!

Reply:
Seems like it worked...? Not sure, the checks are failing cuz of formatting, but i sticked to the same formatting as used originally.

Reply:
please run `cargo fmt` and update the PR so we can see it turn green!

Reply:
Ahh, so there is an autoformatter? Should be good now, hopefully.